resources/scripts.js

```diff
@@ -2,88 +2,184 @@ function firstGame() {
     }
     playerPoint = playerRoll;
     pointOpen = true;
     buttonPosition(playerRoll);
-    displayMessage(`Your point is ${playerPoint}`);
+    buildSummary(SHOWPOINT);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonStates();
+    return;
 }
 
-function gameRoll() {
+async function gameRoll() {
     diceRoll();
-    displayMessage(`You rolled ${playerRoll}`);
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    buildSummary(DICEROLLED);
+    //if (secondaryBetList.length !== 0) {
+    //    secondaryRoll(playerRoll);
+    //}
+    if (testObjPop(secondaryBetListO) === false) {
+        secondaryRoll(playerRoll);
+    }
     if (playerRoll === playerPoint) {
-        playerWin();
+        payOut(true, true, betAmount);
         return;
     }
     if (playerRoll === 7) {
-        playerLose();
+        payOut(true, false);
         return;
     }
-    displayMessage('No action, roll again');
+    buildSummary(NOACTION);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonStates();
+    return;
 }
-
 function checkGameState() {
     if (pointOpen === false) {
-       firstRoll(); 
-       return;
+        firstRoll();
+        return;
     }
     gameRoll();
+    return;
+}
+
+//function secondaryRoll(roll) {
+//    for (let i = 0; i < secondaryBetList.length; i++) {
+//        const currentBet = secondaryBetList[i];
+//        if (currentBet.point === 1) {
+//            if (roll === 7 || roll === 11) {
+//                removeBet(i);
+//                i--;
+//                playerWin(SECONDARY, currentBet.amount);
+//            } else if (roll === 2 || roll === 3 || roll === 12) {
+//                removeBet(i);
+//                i--;
+//                playerLose(SECONDARY);
+//            } else {
+//                currentBet.point = roll;
+//            }
+//        } else {
+//            if (roll === 7) {
+//                removeBet(i);
+//                i--;
+//                playerLose(SECONDARY);
+//            } else if (currentBet.point === roll) {
+//                removeBet(i);
+//                i--;
+//                playerWin(SECONDARY, currentBet.amount);
+//            } else {
+//                // Placeholder, prefer different notification method
+//                displayMessage(NOACTION);
+//            }
+//        }
+//    }
+//    return;
+//}
+
+function secondaryRoll(roll) {
+    if (roll === 7) {
+        for (let i = 4; i < 11; i++) {
+            if (secondaryBetListO[i] > 0) {
+                buildSummary(SECONDARYLOSE, i);
+                payOut(false, false);
+                delete secondaryBetListO[i];
+            }
+        }
+    } else {
+        if (roll === 4 || roll === 5 || roll === 6 || roll === 8 || roll === 9 || roll === 10) {
+            buildSummary(PLACEWIN, roll);
+            payOut(false, true, secondaryBetListO[roll]);
+            delete secondaryBetListO[roll];
+        }
+    }
+    if (secondaryBetListO[1] > 0) {
```

Comment:
NaN issue appears to be here. After Come bet changes to Place, playerMoney becomes NaN.

Reply:
Added 3rd argument to payOut() as this could cause NaN.

Reply:
Completed NaN issue fix.